Sources/Srt/globals.alusus

```diff
@@ -1,3 +1,4 @@
+import "libLLVM.so";
```

Comment:
<div dir="rtl">
@sarmadka @falhumai96 مارأيكم بدمج `libLLVM.so` و `libalusus_spp.so` في الأسس مباشرة

</div>

Reply:
الLLVM كانت جزء من الSPP من الاصل، لكني واجهت مشاكل في بناء الSPP كshared library في الMinGW، مما أدى إلى بنائها كshared library. هنالك خياران لبناء الأسس في ويندوز: ١) بناء SPP statically (سوف يجعل بناء الاسس مهلكاً). ٢) بناء LLVM dynamically، و استدعائها dynamically (الخيار المفضل لدي). المشكلة انهو الويندوز لا يقدر أن يحمل عدد هائل من الexported symbols في الshared library، و يجعل بناء الSPP كshared library مستحيلاً. ما رأيك @sarmadka؟

Reply:
Apologies I will have to type in English, the above Arabic text looks messed up.

So, basically LLVM was part of SPP and compiled statically, but since we are planning to support Windows, and while experimenting with MinGW builds, SPP was not compiling due to the huge number of exported symbols in libalusus_spp generated by LLVM. Windows has a limited number of allowed exported symbols, unlike Linux. So, I was left with two choices, either compile SPP statically (will make me suffer while developing on Windows due to very slow compilation speed everytime I make a very small change), or compile LLVM dynamically, and load it at run time from globals.alusus (still experimenting on this one, but I have a feeling it will be better since LLVM need not to be recompiled).

Reply:
<p dir=rtl>
الLLVM كانت جزء من الSPP من الاصل، لكني واجهت مشاكل في بناء الSPP كshared library في الMinGW، مما أدى إلى بناء الLLVM كshared library. هنالك خياران لبناء الأسس في الويندوز: ١) بناء SPP statically (سوف يجعل بناء الاسس مهلكاً). ٢) بناء LLVM dynamically، و استدعائها dynamically (الخيار المفضل لدي، لكني مازلت اجربه حالياً في الويندوز (لدي حدس ان البناء سوف يكون سهلاً، لأن الLLVM لاتحتاج الى إعادة البناء)). المشكلة ان الويندوز لا يقدر أن يحمل عدد هائل من الexported symbols في الshared library، و يجعل بناء الSPP كshared library  مستحيلاً. ما رأيك في هذا الحل @sarmadka؟
<p>

Reply:
<p dir=rtl>
حل ملائم على ما أظن. سنحتاج فقط لاختباره جيدا للتأكد أن نسختنا من LLVM لن تتعارض مع النسخة الموجودة على النظام إن وجدت. فيما عدا هذا ليس هناك ما يستوجب بقاء LLVM ضمن SPP.
</p>

Reply:
<p dir=rtl>
@xlmnxp لماذا تقترح دمجهما بالأسس؟ إن كنت تقصد توفير عملية 'اشمل' على المستخدم فبإمكاننا فعل ذلك دون الحاجة لدمج المكتبة نفسها. أي بإمكاننا جعل الأسس تحمل المكتبتين تلقائيا.
</p>

Reply:
@sarmadka اخي هذه المكتبات لا يمكن ان يعمل اي تطبيق او برنامج برمج بالأسس بدونها
بمعنى السماح بإمكانية ازالتها خطأ 
من المفترض ان يتم استدعائها ضمنيًا داخل الأسس